pinot-spi/src/main/java/org/apache/pinot/spi/config/table/ingestion/SchemaConformingTransformerV2Config.java

```diff
@@ -0,0 +1,237 @@
+
+  public SchemaConformingTransformerV2Config setIndexableExtrasField(String indexableExtrasField) {
+    _indexableExtrasField = (null == indexableExtrasField) ? _indexableExtrasField : indexableExtrasField;
+    return this;
+  }
+
+  public SchemaConformingTransformerV2Config setEnableUnindexableExtras(Boolean enableUnindexableExtras) {
+    _enableUnindexableExtras = enableUnindexableExtras == null ? _enableUnindexableExtras : enableUnindexableExtras;
+    return this;
+  }
+
+  public String getUnindexableExtrasField() {
+    return _enableUnindexableExtras ? _unindexableExtrasField : null;
+  }
+
+  public SchemaConformingTransformerV2Config setUnindexableExtrasField(String unindexableExtrasField) {
+    _unindexableExtrasField = (null == unindexableExtrasField) ? _unindexableExtrasField : unindexableExtrasField;
+    return this;
+  }
+
+  public String getUnindexableFieldSuffix() {
+    return _unindexableFieldSuffix;
+  }
+
+  public SchemaConformingTransformerV2Config setUnindexableFieldSuffix(String unindexableFieldSuffix) {
+    _unindexableFieldSuffix = (null == unindexableFieldSuffix) ? _unindexableFieldSuffix : unindexableFieldSuffix;
+    return this;
+  }
+
+  public Set<String> getFieldPathsToDrop() {
+    return _fieldPathsToDrop;
+  }
+
+  public SchemaConformingTransformerV2Config setFieldPathsToDrop(Set<String> fieldPathsToDrop) {
+    _fieldPathsToDrop = (null == fieldPathsToDrop) ? _fieldPathsToDrop : fieldPathsToDrop;
+    return this;
+  }
+
+  public Map<String, String> getColumnNameToJsonKeyPathMap() {
+    return _columnNameToJsonKeyPathMap;
+  }
+
+  public SchemaConformingTransformerV2Config setColumnNameToJsonKeyPathMap(
+      Map<String, String> columnNameToJsonKeyPathMap) {
+    _columnNameToJsonKeyPathMap = (null == columnNameToJsonKeyPathMap)
+        ? _columnNameToJsonKeyPathMap : columnNameToJsonKeyPathMap;
+    return this;
+  }
+
+  public String getMergedTextIndexField() {
+    return _mergedTextIndexField;
+  }
+
+  public SchemaConformingTransformerV2Config setMergedTextIndexField(String mergedTextIndexField) {
+    _mergedTextIndexField = (null == mergedTextIndexField) ? _mergedTextIndexField : mergedTextIndexField;
+    return this;
+  }
+
+  public Integer getMergedTextIndexTokenMaxLength() {
+    return _mergedTextIndexTokenMaxLength;
+  }
+
+  public SchemaConformingTransformerV2Config setMergedTextIndexTokenMaxLength(
+      Integer mergedTextIndexTokenMaxLength
+  ) {
+    _mergedTextIndexTokenMaxLength = (null == mergedTextIndexTokenMaxLength)
+        ? _mergedTextIndexTokenMaxLength : mergedTextIndexTokenMaxLength;
+    return this;
+  }
+
+  public Integer getMergedTextIndexShinglingOverlapLength() {
+    return _mergedTextIndexShinglingOverlapLength;
+  }
+
+  public SchemaConformingTransformerV2Config setMergedTextIndexShinglingTokenOverlapLength(
+      Integer mergedTextIndexShinglingOverlapLength) {
+    _mergedTextIndexShinglingOverlapLength = mergedTextIndexShinglingOverlapLength;
```

Comment:
Can you double check what is the side-effect of passing a `null` Integer object and whether we need to check for null value?

Reply:
We are treating `mergedTextIndexShinglingOverlapLength = null` as a special case in transformer, there should not be side effect as long as we check its null value during usage
`if (null == mergedTextIndexShinglingOverlapLength) {
        generateTextIndexToken(kv, luceneTokens, mergedTextIndexTokenMaxLength);
      } else {
        generateShingleTextIndexToken(kv, luceneTokens, mergedTextIndexTokenMaxLength,
            mergedTextIndexShinglingOverlapLength);
      }`